dcore/advection.py

```diff
@@ -122,3 +115,53 @@ def apply(self, x_in, x_out):
         dt = self.state.timestepping.dt
         self.solver.solve()
         x_out.assign(x_in + dt*self.dq)
+
+
+class DGAdvection(Advection):
+
+    def __init__(self, state):
+
+        super(DGAdvection, self).__init__(state)
+
+        dt = state.timestepping.dt
+
+        # by convention the last space is the DG space
+        DGSpace = state.V[-1]
```

Comment:
I note that at the moment, the fully DG space is not the last space in all cases (it's the CP-staggered temperature space for the 3D stuff).

Of course, another option, given that making function spaces is cheap if you've already built one (and all the data are shared), you could just *make* a DG space in this function, then you don't have to worry!